rust/sbd-client/src/raw_client.rs

```diff
@@ -47,6 +63,51 @@ impl WsRawConnect {
             request.headers_mut().insert(k, v);
         }
 
+        if let Some(auth_material) = auth_material {
+            let mut auth_url =
+                url::Url::parse(&full_url).map_err(std::io::Error::other)?;
+            auth_url.set_path("/authenticate");
+            match auth_url.scheme() {
+                "ws" => {
+                    let _ = auth_url.set_scheme("http");
+                }
+                "wss" => {
+                    let _ = auth_url.set_scheme("https");
+                }
+                _ => (),
+            }
+
+            let token = tokio::task::spawn_blocking(move || {
```

Comment:
Why do you spawn a task for this request instead of making the request from the main thread?

Reply:
ureq is a blocking http library. To use it in an async context, you have to use spawn blocking.